sql/blocks/blocks.go

```diff
@@ -0,0 +1,122 @@
+package blocks
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/spacemeshos/go-spacemesh/codec"
+	"github.com/spacemeshos/go-spacemesh/common/types"
+	"github.com/spacemeshos/go-spacemesh/sql"
+)
+
+func decodeBlock(reader io.Reader, id types.BlockID) (*types.Block, error) {
+	inner := types.InnerBlock{}
+	_, err := codec.DecodeFrom(reader, &inner)
+	if err != nil {
+		return nil, fmt.Errorf("failed to decode block %s: %w", id, err)
+	}
+	return types.NewExistingBlock(id, inner), nil
+}
+
+// Add block to the database.
+func Add(db sql.Executor, block *types.Block) error {
+	bytes, err := codec.Encode(block.InnerBlock)
+	if err != nil {
+		return fmt.Errorf("encode %w", err)
+	}
+	if _, err := db.Exec("insert into blocks (id, layer, block) values (?1, ?2, ?3);",
+		func(stmt *sql.Statement) {
+			stmt.BindBytes(1, block.ID().Bytes())
+			stmt.BindInt64(2, int64(block.LayerIndex.Value))
+			stmt.BindBytes(3, bytes) // this is actually should encode block
+		}, nil); err != nil {
+		return fmt.Errorf("insert %s: %w", block.ID(), err)
+	}
+	return nil
+}
+
+// Has a block in the database.
+func Has(db sql.Executor, id types.BlockID) (bool, error) {
+	rows, err := db.Exec("select 1 from blocks where id = ?1;",
+		func(stmt *sql.Statement) {
+			stmt.BindBytes(1, id.Bytes())
+		}, nil,
+	)
+	if err != nil {
+		return false, fmt.Errorf("has ballot %s: %w", id, err)
+	}
+	return rows > 0, nil
+}
+
+// Get block with id from database.
+func Get(db sql.Executor, id types.BlockID) (rst *types.Block, err error) {
+	if rows, err := db.Exec("select block from blocks where id = ?1;", func(stmt *sql.Statement) {
+		stmt.BindBytes(1, id.Bytes())
+	}, func(stmt *sql.Statement) bool {
+		rst, err = decodeBlock(stmt.ColumnReader(0), id)
+		return true
+	}); err != nil {
+		return nil, fmt.Errorf("get %s: %w", id, err)
+	} else if rows == 0 {
+		return nil, fmt.Errorf("%w block %s", sql.ErrNotFound, id)
+	}
+	return rst, err
+}
+
+// SetValid updates verified status for a block.
+func SetValid(db sql.Executor, id types.BlockID) error {
+	return setValidity(db, id, 1)
+}
+
+// SetInvalid updates blocks to an invalid status.
+func SetInvalid(db sql.Executor, id types.BlockID) error {
+	return setValidity(db, id, -1)
```

Comment:
put 1 and -1 as named constants? 
btw, why -1 instead of 0?

Reply:
0 is a default value and means undecided

Reply:
I'd also prefer having named constants